### PR TITLE
chore(docs): document migratioon process

### PR DESCRIPTION
**Pull request by kevinslin**
Add documentation on how `dendorn.yml` configuration is changed during upgrades

### CODE REVIEW

**Comment by kevinslin** [comment]
See https://github.com/dendronhq/dendron-docs/commit/72d2a698992a0187d358287e96d554ae6f0a89d1 for more docs

File: packages/plugin-core/src/_extension.ts [hunk 1]
```diff
@@ -427,6 +427,7 @@ export async function _activate(
       } else {
         // no migration changes.
         // see if we need to force a config migration.
+        // see [[Run Config Migration|dendron://dendron.docs/pkg.dendron-engine.t.upgrade.arch.lifecycle#run-config-migration]]
```
Comment (SeriousBug) [inline packages/plugin-core/src/_extension.ts:430]: Should be `...#run-migration]]`

**Review by SeriousBug — APPROVED** [review]
Looks good, just one broken anchor.

Also, not related to this PR but there's a link for `[[2021 06 Deprecate Workspace Settings|proj.2021-06-deprecate-workspace-settings]]` in `_extension.ts`, which is actually in `private` vault which is not included in the workspace. We should move the note if we want it accessible.

File: packages/plugin-core/src/_extension.ts (hunk 1 above)
Comment (kevinslin) [reply packages/plugin-core/src/_extension.ts:430]: fixed

**Pull request merged**